.pre-commit-config.yaml

```diff
@@ -0,0 +1,15 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.3.0
+    hooks:
+    - id: check-yaml
+    - id: end-of-file-fixer
+    - id: trailing-whitespace
+  - repo: local
+    hooks:
+      - id: cargo-fmt
+        name: Format Rust
+        description: Reformat Rust code as needed
+        entry: bash -c 'cd rust && cargo fmt'
```

Comment:
Check that cargo fmt returns nonzero if changes are made.
- If it returns zero (success) even when changes are made, is there an argument to make it fail?

Otherwise, I've been using this command in my pre-commit, that just checks the format is right:

```shell
cargo fmt --manifest-path=rust/Cargo.toml --all --check
```

Reply:
Also consider adding `--all` so that the check runs for all crates in the workspace.